src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/controller/CertificateApi.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.bpdmcertificatemanagement.controller
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import org.eclipse.tractusx.bpdmcertificatemanagement.dto.request.CertificateDocumentRequestDto
+import org.eclipse.tractusx.bpdmcertificatemanagement.dto.response.CertificateDocumentResponseDto
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.service.annotation.HttpExchange
+import org.springframework.web.service.annotation.PostExchange
+
+@RequestMapping("/api/catena", produces = [MediaType.APPLICATION_JSON_VALUE])
+@HttpExchange("/api/catena")
+interface CertificateApi {
+
+    @Operation(
+        summary = "Provide a specific certificate document for a given BPN.",
+        operationId = "setCertificateDocument",
+        description = "Use this API call to provide a specific certificate document for a given BPN.",
+        responses = [
+            ApiResponse(responseCode = "200", description = "Document for the given ID"),
+            ApiResponse(responseCode = "400", description = "Malformed URL", content = [Content()]),
+            ApiResponse(responseCode = "401", description = "Unauthorized", content = [Content()]),
+            ApiResponse(responseCode = "404", description = "BPN not found", content = [Content()]),
```

Comment:
I did similar things before but now i'm not so sure about that even though it's rest conform. When people receive a 404 they immediately think that there is a wrong path. So in this case when a path is not found and when a BPN is not found is the same error.

I think I wouldn't use 404 anymore like this

Reply:
Which httpStatus code is relevant then, May i go with 422("Unprocessable Entity")?

Reply:
Oh, that's a rare code but I think it is very fitting actually. I think it's a good idea. Maybe its also something to look at for BPDM actually